From: "Barak A. Pearlmutter" <barak+git@pearlmutter.net>
Date: Mon, 31 Aug 2015 20:18:42 +0100
Subject: reenable site-init

The upstream build scripts invoke emacs with the -Q option, --quick.
As documented,

  $ man emacs | egrep -A1 '[-][qQ]|site-file'
		-q, --no-init-file
			Do not load an init file.
  --
		--no-site-file
			Do not load the site-wide startup file.
  --
		-Q, --quick
			Similar to "-q --no-site-file --no-splash".  Also, avoid processing X resources.

this does not load the site initialization files.  But in Debian, the
site initialization file needs to be loaded, in order to allow access
to other elisp packages.  This patch uses the -q option instead, so
the user-specific init file is not loaded, but the site one is.

Forwarded: no
Author: Barak A. Pearlmutter <barak@pearlmutter.net>
---
 Documentation/Makefile | 6 +++---
 Makefile               | 2 +-
 default.mk             | 2 +-
 3 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/Documentation/Makefile b/Documentation/Makefile
index da3b1f5..aae00d5 100644
--- a/Documentation/Makefile
+++ b/Documentation/Makefile
@@ -23,15 +23,15 @@ texi: $(TEXIPAGES)
 	@printf "WARNING: Discard these whitespace errors before committing.\n"
 	@printf "WARNING: Also see https://github.com/magit/magit/issues/1912.\n"
 	@printf "Generating magit.texi\n"
-	@$(EMACSBIN) -batch -Q $(DOC_LOAD_PATH) \
+	@$(EMACSBIN) -batch -q $(DOC_LOAD_PATH) \
 	-l ox-texinfo+.el magit.org -f org-texinfo+export-to-texinfo
 	@echo >> magit.texi
 	@printf "Generating magit-popup.texi\n"
-	@$(EMACSBIN) -batch -Q $(DOC_LOAD_PATH) \
+	@$(EMACSBIN) -batch -q $(DOC_LOAD_PATH) \
 	-l ox-texinfo+.el magit-popup.org -f org-texinfo+export-to-texinfo
 	@echo >> magit-popup.texi
 	@printf "Generating with-editor.texi\n"
-	@$(EMACSBIN) -batch -Q $(DOC_LOAD_PATH) \
+	@$(EMACSBIN) -batch -q $(DOC_LOAD_PATH) \
 	-l ox-texinfo+.el with-editor.org -f org-texinfo+export-to-texinfo
 	@echo >> with-editor.texi
 
diff --git a/Makefile b/Makefile
index 4244ad7..5770ea2 100644
--- a/Makefile
+++ b/Makefile
@@ -77,7 +77,7 @@ test:
 	(ert-run-tests-batch-and-exit))"
 
 test-interactive:
-	@$(EMACSBIN) -Q $(LOAD_PATH) --eval "(progn\
+	@$(EMACSBIN) -q $(LOAD_PATH) --eval "(progn\
 	(load-file \"t/magit-tests.el\")\
 	(ert t))"
 
diff --git a/default.mk b/default.mk
index 4309ca5..8b2598a 100644
--- a/default.mk
+++ b/default.mk
@@ -70,7 +70,7 @@ else
   LOAD_PATH ?= -L $(TOP)/lisp -L $(DASH_DIR)
 endif
 
-BATCH = $(EMACSBIN) -batch -Q $(LOAD_PATH)
+BATCH = $(EMACSBIN) -batch -q $(LOAD_PATH)
 
 EMACSOLD := $(shell $(BATCH) --eval \
   "(and (version< emacs-version \"$(EMACS_VERSION)\") (princ \"true\"))")
